src/components/pages/MainQuizPage/MainQuizPage.tsx

```diff
@@ -1,23 +1,54 @@
 import { QuestionContainer } from "./QuestionContainer/QuestionContainer";
 import { EndQuizButton } from "./EndQuizButton/EndQuizButton";
 import { ProgressBarContainer } from "./ProgressBarContainer/ProgressBarContainer";
-import { TimerContainer } from "./TimerContainer/TimerContainer";
-import { initialQuestion } from "../../initialValues/initialQuestion";
+import { questions } from "../../initialValues/initialQuestion";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { RESULT } from "../../Router/routes";
 
 const MainQuizPage = () => {
+
+    const [activeQuestionIndex, setActiveQuestionIndex] = useState(0);
+    const [seconds, setSeconds] = useState(20);
+    const navigate = useNavigate();
+
+    const minutesString = String(Math.floor(seconds / 60)).padStart(2, '0');
+    const secondsString = String(seconds % 60).padStart(2, "0");
+    const isDanger = seconds < 10;
+
+    useEffect(() => {
+        const timerId = setInterval(() => setSeconds((prev) => Math.max(prev - 1, 0)), 1000);
+        return () => {
+            clearInterval(timerId);
+        };
+    }, []);
+
+    if (seconds <= 0) navigate(RESULT);
+
     return (
         <div className="main-quiz-page page-container">
             <QuestionContainer
-                question={initialQuestion}
+                question={questions[activeQuestionIndex]}
+                changeQuestion={() => {
+                    activeQuestionIndex === questions.length - 1 ?
+                        navigate(RESULT) :
+                        setActiveQuestionIndex((prev) => prev + 1)
+                }}
```

Comment:
1. please extract a callback into a constant
2. change `changeQuestion` to `onChangeQuestion` according to the convention